sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -86,12 +86,18 @@ abstract class Expression extends TreeNode[Expression] {
       case (i1, i2) => i1 == i2
     }
   }
+
+  /**
+   * Check the input data types, returns `TypeCheckResult.success` if it's valid,
+   * or return a `TypeCheckResult` with an error message if invalid.
+   */
+  def checkInputDataTypes(): TypeCheckResult = TypeCheckResult.success
```

Comment:
Then we need to implement it for a lot of expressions and decide error message for them in this PR... Can we do it step by step and finally remove the default implementation?